cmd/template/kubernetes.go

```diff
@@ -43,8 +43,8 @@ func buildTemplateKubernetesCore() *yaml.Node {
 		"strategy1", "A named strategy from the strategies block. This example uses the name strategy1.")...)
 
 	constraintNode, constraintValuesNode := util.BuildMapNode("constraints", "")
-	dependsOnNode, dependsOnValuesNode := util.BuildSequenceNode("dependsOn", "Defines the deployments that must reach a successful state (defined as status == SUCCEEDED) before this deployment can start.Deployments with the same dependsOn criteria will execute in parallel.")
-	beforeNode, beforeValuesNode := util.BuildSequenceNode("beforeDeployment", "A set of steps that are executed in parallel")
+	dependsOnNode, dependsOnValuesNode := util.BuildSequenceNode("dependsOn", "Defines the deployments that must complete successfully before this deployment can start. Deployments with the same dependsOn criteria execute in parallel.")
+	beforeNode, beforeValuesNode := util.BuildSequenceNode("beforeDeployment", "Conditions that must be met before the deployment can start. They execute in parallel.")
```

Comment:
Since the "beforeDeployment" constraints are limited to specific types should we explain that? Currently we support pause, either for a set time or until approved.

Reply:
Hmmm, how would you feel about adding a line like? 

"Configure pause behavior using the same format as the pause step in the strategies block."

I'm assuming we're eventually going to add more options, so listing them all out as an inline comment would be gnarly. 

We could also toss in a link to the docs. Actually, we should probably toss in a links to the reference page for this at the top of the template.

Thoughts?

Reply:
This needs some more noodling on my part, so I'm gonna merge this PR and take another pass on it since it'll be in multiple templates.